Add Interaction Localization support

OurTwinkie opened the pull request:
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing
[pull-request]: https://github.com/JDA-Applications/JDA-Utilities/pulls

## Pull Request

#### Pull Request Checklist
Please follow the following steps before opening this PR.<br>
PRs that do not complete the checklist will be subject to denial for
missing information.

- [ ] I have checked the [pull request page][pull-request] for upcoming
      or merged features/bug fixes.
- [ ] I have read JDA's [contributing guidelines][contributing].

#### Pull Request Information
Check and fill in the blanks for all that apply:

- [ ] My PR fixes a bug, error, or other issue with the library's codebase.
- [ ] My PR is for the `______` module of the JDA-Utilities library.
- [ ] My PR creates a new module for the JDA-Utilities library: `______`.

#### Description

Describe your PR here.

OurTwinkie committed:
Change JDA version 18 -> 19

OurTwinkie committed:
Change JDA version 18 -> 19

OurTwinkie committed:
Change JDA version 18 -> 19

OurTwinkie committed:
Change JDA version 18 -> 19

OurTwinkie committed:
Change JDA version 18 -> 19

Chew committed:
Optimize imports and comments

OurTwinkie committed:
Merge branch 'Chew:master' into master

OurTwinkie committed:
Add name and description localization for the slash command

Chew commented on command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:
```diff
@@ -72,6 +68,11 @@
  */
 public abstract class SlashCommand extends Command
 {
+
```
Docs

Chew commented on command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:
```diff
@@ -72,6 +68,11 @@
 {
+
+    protected Map<DiscordLocale, String> nameLocalization = new HashMap<>();
+
```
Docs

Chew commented on command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:
```diff
@@ -505,4 +517,12 @@ else if(cooldownScope.equals(CooldownScope.GUILD) && event.getGuild()==null)
         else
             return front+" "+cooldownScope.errorSpecification+"!";
     }
+
```
Docs

Chew commented on command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:
```diff
@@ -505,4 +517,12 @@ else if(cooldownScope.equals(CooldownScope.GUILD) && event.getGuild()==null)
+    public Map<DiscordLocale, String> getNameLocalization() {
+        return nameLocalization;
+    }
+
```
Docs

Chew reviewed (changes requested):
Great work! I'll need to do some testing then we'll be good to go.

I may want a helper to help make the locales, but we'll see.

Chew commented:
Do menus also have locales?

OurTwinkie commented:
> Do menus also have locales?

No, today I will do the same for the menu and write the docs.

OurTwinkie committed:
Add Context Menu Localization support, add docs to SlashCommand and ContextMenu

Chew reviewed (approved):
Seems good! Gonna do some testing just to be sure.

Chew commented:
<img width="257" alt="image" src="https://user-images.githubusercontent.com/8278263/194763195-07e50557-c623-4e49-9df3-e9497bb78de3.png">
Weird, I added the name localization to a parent command, but it didn't take effect for the child. Not sure what's up. Nor did description locale take effect.

OurTwinkie committed:
Fix subcommands localization

OurTwinkie commented:
> <img alt="image" width="257" src="https://user-images.githubusercontent.com/8278263/194763195-07e50557-c623-4e49-9df3-e9497bb78de3.png">
> 
> Weird, I added the name localization to a parent command, but it didn't take effect for the child. Not sure what's up. Nor did description locale take effect.

Fixed it, now it works.

Chew commented:
<img width="238" alt="image" src="https://user-images.githubusercontent.com/8278263/196039682-8064d94d-0fad-4f3b-b121-b994a3e6fa01.png">
Looks great!
